### PR TITLE
Exchange rates fetch and update in app

### PR DESCRIPTION
**Pull request by NovacAndrei**

### CODE REVIEW

File: src/app.tsx [hunk 1]
```diff
@@ -131,6 +133,12 @@ export default class App extends React.Component<{}, IState> {
                 if (!this.reduxStateLoaded) {
                     this.reduxStateLoaded = true;
 
+                    fetchExchangeRates((exchangeRates: { [tokenType: string]: number }) => {
```
Comment (krisboit) [inline src/app.tsx]: rename this to `subscribeExchangeRates`

File: src/core/utils/balance.ts [hunk 2]
```diff
@@ -50,20 +50,7 @@ export const convertAmount = (
...
-            const avTokens = Object.keys(exchangeRates[fromToken]);
-            for (const avToken of avTokens) {
-                if (exchangeRates[avToken] && exchangeRates[avToken][toToken]) {
-                    return amount
-                        .multipliedBy(exchangeRates[fromToken][avToken])
-                        .multipliedBy(exchangeRates[avToken][toToken]);
-                }
-            }
-        }
+        return amount.multipliedBy(exchangeRates[fromToken]);
```
Comment (krisboit) [inline src/core/utils/balance.ts]: this is ok for USD :)) what about EUR ?

File: src/core/utils/exchange-rates.ts [hunk 3]
```diff
@@ -0,0 +1,7 @@
+import firebase from 'react-native-firebase';
+
+export const fetchExchangeRates = (callback: any) => {
```
Comment (krisboit) [inline src/core/utils/exchange-rates.ts]: we need to check the performance with very large set of data. like exchange rates for 2k tokens.

File: src/core/utils/exchange-rates.web.ts [hunk 4]
```diff
@@ -0,0 +1 @@
+export const fetchExchangeRates = (callback: any) => callback({});
```
Comment (krisboit) [inline src/core/utils/exchange-rates.web.ts]: we need exchange rates on web too :P

File: src/screens/dashboard/dashboard.tsx [hunk 5]
```diff
@@ -56,7 +56,7 @@ const mapStateToProps = (state: IReduxState) => ({
     blockchains: getBlockchains(state),
     selectedBlockchain: getSelectedBlockchain(state),
     selectedAccount: getSelectedAccount(state),
-    exchangeRates: (state as any).market.exchangeRates,
+    exchangeRates: state.market.exchangeRates,
```
Comment (krisboit) [inline src/screens/dashboard/dashboard.tsx]: are exchange rates needed on dashboard ?

Comment (NovacAndrei) [reply src/screens/dashboard/dashboard.tsx]: image

`exchangeRates` are being used on Dashboard in order to `calculateBalance` for `CoinBalanceCard`

Comment (krisboit) [reply src/screens/dashboard/dashboard.tsx]: good, that's fine :)
but on the screenshot i see a problem : `toCurrency="USD"`, please change it, use the value from redux

Comment (NovacAndrei) [reply src/screens/dashboard/dashboard.tsx]: ok sure, fixing this

File: src/core/utils/exchange-rates.web.ts (hunk 4 above)
Comment (NovacAndrei) [reply src/core/utils/exchange-rates.web.ts]: TODO: another task for web firebase

File: src/core/utils/balance.ts (hunk 2 above)
Comment (NovacAndrei) [reply src/core/utils/balance.ts]: Done

**Pull request merged**